backend/crud.py

```
@@ -0,0 +1,245 @@
+
+
+def verify_password(plain_password: str, hashed_password: str) -> bool:
+    """パスワードを検証"""
+    return pwd_context.verify(plain_password, hashed_password)
+
+
+def update_user_coins(db: Session, user_id: UUID, coin_delta: int) -> models.User:
+    """ユーザーのコインを更新"""
+    user = get_user_by_id(db, user_id)
+    if not user:
+        raise ValueError(f"user {user_id} not found")
```

Ruff (0.14.7)

Avoid specifying long messages outside the exception class

(TRY003)